File: app/src/main/java/com/evento/apptallerautoridad/network/ApiService.java
package com.evento.apptallerautoridad.network;

import com.evento.apptallerautoridad.data.entity.Expositor;
import com.evento.apptallerautoridad.data.entity.Agenda;
import com.evento.apptallerautoridad.data.entity.Hotel;

import kikopalomares.com.apicallsretrofit.data.remote.response.AgendaResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ApiService {


    @GET("/expositor/id/{idExpositor}")
    Call<Expositor> obtenerExpositor(@Path("idExpositor") String idExpositor);


    @GET("/agenda/id/{idAgenda}")
    Call<Agenda> obtenerAgenda(@Path("idAgenda") String idAgenda);

   // @GET("/usuario/catalogo/{codCatalogo}/{codGrup}")
    //Call<GruposResponse> obtenerGrupos(@Path("codGrup") String codCatalogo, @Path("codGrup") String codGrup);

    @GET("/hotel/codigo/{codigo}")
    Call<Hotel> obtenerHotel(@Path("codigo") String id);


}

File: app/src/main/java/com/evento/apptallerautoridad/data/ApiService.kt
package kikopalomares.com.apicallsretrofit.data.remote


import kikopalomares.com.apicallsretrofit.data.remote.response.ExpositoresResponse
import kikopalomares.com.apicallsretrofit.data.remote.response.AgendaResponse
import kikopalomares.com.apicallsretrofit.data.remote.response.HotelResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path


interface ApiService {

    @GET("id/{idExpositor}")
    fun getAllPosts(@Path("idExpositor") id : String): Call<ExpositoresResponse>


    @GET("id/{idAgenda}")
    fun getAllAgenda(@Path("idAgenda") id : String): Call<AgendaResponse>

    @GET("codigo/{codigo}")
    fun getAllHotel(@Path("codigo") codigo : String): Call<HotelResponse>


}


